### PR TITLE
Preact spike

### PR DESCRIPTION
**Pull request by liywjl**
## What does this change?
Move away from React to Preact

## Why?
Reduces the bundle size dramatically

## Link to supporting Trello card
https://trello.com/c/FlHadfnk/1111-spike-preact

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,19 +76,20 @@
...
         "prettier-eslint": "^9.0.0",
         "pretty-bytes": "^4.0.2",
-        "react": "^16.8.6",
-        "react-dom": "^16.8.6",
         "reset-css": "^5.0.1",
         "sanitize-html": "^1.20.1"
     },
     "devDependencies": {
         "@babel/core": "^7.5.5",
+        "@babel/helper-create-regexp-features-plugin": "^7.8.3",
```
Comment (oliverlloyd) [inline package.json]: What does this do?

**Comment by gtrufitt** [comment]
Great! This looks positive, particularly in bundle size. I'd be interested in a like-for-like comparison of load times, server and client. 

Are you able to get some metrics together before we merge on things like when the islands mount on the client side (might need to add some usertiming metrics, but CODE is accessible to WPT) and how long a request takes on the server?

We've got some perf endpoints already which you could run locally (maybe?) - you'd have to run the 'prod' server using `make start` and then `make monitor` to see the logs. The endpoints are in https://github.com/guardian/dotcom-rendering/blob/master/src/app/server.ts - though I'm not quite sure how it works, maybe @nicl can help!

I don't expect it to make a huge amount of difference but it'd be good to have visibility on it before it's live.

**Comment by liywjl** [comment]
> Great! This looks positive, particularly in bundle size. I'd be interested in a like-for-like comparison of load times, server and client.
> 
> Are you able to get some metrics together before we merge on things like when the islands mount on the client side (might need to add some usertiming metrics, but CODE is accessible to WPT) and how long a request takes on the server?
> 
> We've got some perf endpoints already which you could run locally (maybe?) - you'd have to run the 'prod' server using `make start` and then `make monitor` to see the logs. The endpoints are in https://github.com/guardian/dotcom-rendering/blob/master/src/app/server.ts - though I'm not quite sure how it works, maybe @nicl can help!
> 
> I don't expect it to make a huge amount of difference but it'd be good to have visibility on it before it's live.

I have been trying to get some analytics locally but not really able, would be great if we could pair on it

**Comment by liywjl** [comment]
Currently our local webpack dev server doesnt minify our bundle and running tests using lighthouse cause a lot of variation. Therefore I decided to try get some metrics locally:

Screenshot 2020-02-12 at 14 55 37

This compares master to Preact and Webpack branches. Each test was run 4 times, in incognito mode (so without browser addons), and the number here is the average of each metric.

I think there is still a benefit to merging this PR and we do see an increased uptick.

**Pull request merged**